sys/page_revisions.js

```diff
@@ -301,11 +301,14 @@ PRS.prototype.fetchAndStoreMWRevision = function(hyper, req) {
             }
         })
         .then(function(res) {
-            var sameRev = res && res.body.items
-                    && res.body.items.length > 0
-                    && self._checkSameRev(revision, res.body.items[0]);
-            if (!sameRev) {
-                throw new HTTPError({ status: 404 });
+            if (res && res.body.items && res.body.items.length > 0) {
+                var storedRev = res.body.items[0];
+                // The redirect in MW API is based on the latest revision,
+                // so for older revisions it must never be updated.
```

Comment:
Perhaps add a TODO about this not being accurate for old revisions (might miss redirects if latest version is no longer a redirect).